src/zenml/steps/utils.py

```diff
@@ -589,3 +594,72 @@ def Do(
         )
         with fileio.open(self._context.executor_output_uri, "wb") as f:
             f.write(executor_output.SerializeToString())
+
+
+def clone_step(step: Type["BaseStep"], step_name: str) -> Type["BaseStep"]:
```

Comment:
I was also quite intrigued by this problem. I understand that we must not allow anyone to reuse an instance of a step within a pipeline but at the same time, it feels like they should be able to create different instances of the same step definition and use them within the same pipeline. I think cloning is a good solution however it requires that our users take one more action when designing their pipelines, and I was wondering whether we can remove the root of the problem?

Reply:
The limitation comes from TFX and the way that they identify pipeline steps (nodes) by their class name. Basically, every TFX step needs a distinct, resolvable class name. This also makes it very difficult to do the cloning under-the-hood, because we have to dynamically create consistent class names and add them to some module namespace where they can be loaded by TFX.

In other words: TFX is the root (of all) problem(s)

Reply:
I'd love ideas on how we can improve and even hide this. I couldn't find a way. Even writing this utility took a lot of trial-and-error iterations to get right.